modules/data.remote/inst/gee2pecan_l8.py

```diff
@@ -0,0 +1,191 @@
+"""
+Extracts Landsat 8 surface reflactance band data from Google Earth Engine and saves it in a netCDF file
+
+Requires Python3
+
+Bands retrieved: B1, B2, B3, B4, B5, B6, B7, B10, B11 along with computed NDVI
+
+If ROI is a Point, this function can be used for getting SR data from Landsat 7, 5 and 4 as well.
+
+Author: Ayush Prasad
+"""
+from gee_utils import create_geo, get_siteaoi, get_sitename
+import ee
+import pandas as pd
+import geopandas as gpd
+import datetime
+import os
+import xarray as xr
+import numpy as np
+import re
+
+ee.Initialize()
+
+
+def gee2pecan_l8(geofile, outdir, start, end, qc=1):
+    """
+    Extracts Landsat 8 SR band data from GEE
+
+    Parameters
+    ----------
+    geofile (str) -- path to the file containing the name and coordinates of ROI, currently tested with geojson. 
+    
+    outdir (str) -- path to the directory where the output file is stored. If specified directory does not exists, it is created.
+  
+    start (str) -- starting date of the data request in the form YYYY-MM-DD
+    
+    end (str) -- ending date areaof the data request in the form YYYY-MM-DD
+
+    ic (str) -- image collection id of the Landsat SR product from Google Earth Engine
+
+    vi (bool) -- set to True if NDVI needs to be calculated
+
+    qc (bool) -- uses the cloud masking function if set to True
+
+    bands (list of str) -- bands to be retrieved. Default: B5, B4
+
+    Returns
+    -------
+    Nothing:
+            output netCDF is saved in the specified directory. 
+ 
+    """
+
+    # scale (int) Default: 30
+    scale = 30
+    # bands retrieved
+    bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7", "B10", "B11"]
+
+    def reduce_region(image):
+        """
+        Reduces the selected region
+        currently set to mean, can be changed as per requirements.
+        """
+        stat_dict = image.reduceRegion(ee.Reducer.mean(), geo, scale)
+        sensingtime = image.get("SENSING_TIME")
+        return ee.Feature(None, stat_dict).set("sensing_time", sensingtime)
+
+    def mask(image):
+        """
+        Masks clouds and cloud shadows using the pixel_qa band
+        Can be configured as per requirements 
+        """
+        clear = image.select("pixel_qa")
+        return image.updateMask(clear)
+
+    # create image collection depending upon the qc choice
+    if qc == True:
+        landsat = (
+            ee.ImageCollection("LANDSAT/LC08/C01/T1_SR")
+            .filterDate(start, end)
+            .map(mask)
+            .sort("system:time_start", True)
+        )
+
+    else:
+        landsat = (
+            ee.ImageCollection("LANDSAT/LC08/C01/T1_SR")
+            .filterDate(start, end)
+            .sort("system:time_start", True)
+        )
+
+    def calcNDVI(image):
```

Comment:
also this probably belongs to gee_utils with an additional argument that specifies which bands to use

Although I know you don't use it for S2 yet